ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/productcarousel/v1/productcarousel/clientlibs/js/product-carousel.js

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+(function() {
+
+  /**
+   *  Product Carousel Component  
+   */
+  class ProductCarousel {
+
+    // Initialise the variables
+    static selectors = {
+
+      self: "[data-comp-is=productcarousel]",
```

Comment:
Data attribute added for root element

Reply:
The correct format of the selector, according to the spec is `[data-comp-is='productcarousel']` (attribute value between single or double quotes).

Reply:
format corrected